e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -545,6 +545,112 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
+        public static async Task<PublishedNodesEntryModel> CreateSingleNodeModelAsync(IIoTMultipleNodesTestContext context, CancellationTokenSource cts) {
+            IDictionary<string, PublishedNodesEntryModel> simulatedPublishedNodesConfiguration =
+                new Dictionary<string, PublishedNodesEntryModel>(0);
+
+            // With the nested edge test servers don't have public IP addresses and cannot be accessed in this way
+            if (context.IoTEdgeConfig.NestedEdgeFlag != "Enable") {
+                simulatedPublishedNodesConfiguration =
+                    await GetSimulatedPublishedNodesConfigurationAsync(context, cts.Token);
+            }
+
+            PublishedNodesEntryModel model;
+            if (simulatedPublishedNodesConfiguration.Count > 0) {
+                model = simulatedPublishedNodesConfiguration[simulatedPublishedNodesConfiguration.Keys.First()];
+            }
+            else {
+                var opcPlcIp = context.OpcPlcConfig.Urls.Split(TestConstants.SimulationUrlsSeparator)[0];
+                model = new PublishedNodesEntryModel {
+                    EndpointUrl = $"opc.tcp://{opcPlcIp}:50000",
+                    UseSecurity = false,
+                    OpcNodes = new OpcUaNodesModel[] {
+                        new OpcUaNodesModel {
+                            Id = "ns=2;s=SlowUInt1",
+                            OpcPublishingInterval = 10000
+                        }
+                    }
+                };
+            }
+
+            // We want to take one of the slow nodes that updates each 10 seconds.
+            // To make sure that we will not have missing values because of timing issues,
+            // we will set publishing and sampling intervals to a lower value than the publishing
+            // interval of the simulated OPC PLC. This will eliminate false-positives.
+            model.OpcNodes = model.OpcNodes
+                .Where(node => !node.Id.Contains("bad", StringComparison.OrdinalIgnoreCase))
+                .Where(opcNode => opcNode.Id.Contains("SlowUInt"))
+                .Take(1).Select(opcNode => {
+                    var opcPlcPublishingInterval = opcNode.OpcPublishingInterval;
+                    opcNode.OpcPublishingInterval = opcPlcPublishingInterval / 2;
+                    opcNode.OpcSamplingInterval = opcPlcPublishingInterval / 4;
+                    return opcNode;
+                })
+                .ToArray();
+
+            return model;
+        }
+
+        /// <summary>
+        /// Create a multiple nodes model with opcplc nodes
+        /// </summary>
+        /// <param name="IIoTMultipleNodesTestContext">context</param>
+        /// <param name="CancellationTokenSource">cancellation token</param>
+        public static async Task<PublishedNodesEntryModel> CreateMultipleNodesModelAsync(IIoTMultipleNodesTestContext context, CancellationTokenSource cts) {
+            await context.LoadSimulatedPublishedNodes(cts.Token);
```

Comment:
Do we need to call this method in the beginning of `CreateSingleNodeModelAsync()` method as well ?

Reply:
No this is just needed for multiple nodes. For single node it just publish a node with nodeId = `ns=2;s=SlowUInt1`